Cabal/Distribution/PackageDescription/Parsec/FieldDescr.hs

```diff
@@ -0,0 +1,579 @@
+{-# LANGUAGE OverloadedStrings #-}
+-- | 'GenericPackageDescription' Field descriptions
+module Distribution.PackageDescription.Parsec.FieldDescr (
+    -- * Package description
+    pkgDescrFieldDescrs,
+    storeXFieldsPD,
+    -- * Library
+    libFieldDescrs,
+    storeXFieldsLib,
+    -- * Executable
+    executableFieldDescrs,
+    storeXFieldsExe,
+    -- * Test suite
+    TestSuiteStanza (..),
+    emptyTestStanza,
+    testSuiteFieldDescrs,
+    storeXFieldsTest,
+    validateTestSuite,
+    -- * Benchmark
+    BenchmarkStanza (..),
+    emptyBenchmarkStanza,
+    benchmarkFieldDescrs,
+    storeXFieldsBenchmark,
+    validateBenchmark,
+    -- * Flag
+    flagFieldDescrs,
+    -- * Source repository
+    sourceRepoFieldDescrs,
+    -- * Setup build info
+    setupBInfoFieldDescrs,
+    ) where
+
+import           Distribution.Compat.Prelude
+import           Prelude ()
+
+import           Text.PrettyPrint                      (vcat)
+
+import qualified Data.ByteString                       as BS
+import           Data.List                             (dropWhileEnd)
+import qualified Distribution.Compat.Parsec            as Parsec
+import           Distribution.Compiler                 (CompilerFlavor (..))
+import           Distribution.ModuleName               (ModuleName)
+import           Distribution.Package
+import           Distribution.PackageDescription
+import           Distribution.Parsec.Class
+import           Distribution.Parsec.Types.Common
+import           Distribution.Parsec.Types.FieldDescr
+import           Distribution.Parsec.Types.ParseResult
+import           Distribution.PrettyUtils
+import           Distribution.Simple.Utils             (fromUTF8BS)
+import           Distribution.Text                     (disp, display)
+
+-------------------------------------------------------------------------------
+-- common FieldParsers
+-------------------------------------------------------------------------------
+
+-- | This is /almost/ @'many' 'Distribution.Compat.Parsec.anyChar'@, but it
+--
+-- * trims whitespace from ends of the lines,
+--
+-- * converts lines with only single dot into empty line.
```

Comment:
Don't know for sure in the old parser, you'd just have to try it. If so it'd probably be ".." or ". ."